### PR TITLE
Compile SCALAPACK with OpenBLAS32

### PR DESCRIPTION
**Pull request by amontoison**

### CODE REVIEW

File: S/SCALAPACK32/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+mkdir build && cd build
+cmake .. "${CMAKE_FLAGS[@]}"
+
+make -j${nproc} all
+make install
+"""
+
+# OpenMPI and MPICH are not precompiled for Windows
+# Can't get the code to build for PowerPC with libgfortran3
+platforms = expand_gfortran_versions(filter!(p -> !Sys.iswindows(p), supported_platforms(; experimental=true)))
```
Comment (giordano) [inline S/SCALAPACK32/build_tarballs.jl:62]: Bit simpler :slightly_smiling_face: 
```suggestion
platforms = expand_gfortran_versions(supported_platforms(; exclude=Sys.iswindows))
```

Comment (amontoison) [reply S/SCALAPACK32/build_tarballs.jl:62]: `experimental=true` is the default ?

Comment (giordano) [reply S/SCALAPACK32/build_tarballs.jl:62]: It is:  #4160

File: S/SCALAPACK32/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,79 @@
+using BinaryBuilder
+
+name = "SCALAPACK32"
+version = v"2.1.0"
+
+sources = [
+  ArchiveSource("http://www.netlib.org/scalapack/scalapack-2.1.0.tgz",
```
Comment (jeremiahpslewis) [inline S/SCALAPACK32/build_tarballs.jl:7]:
```suggestion
  ArchiveSource("http://www.netlib.org/scalapack/scalapack-$(version).tgz",
```

File: S/SCALAPACK32/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                "61d9216cf81d246944720cfce96255878a3f85dec13b9351f1fa0fd6768220a6"),
+  DirectorySource("./bundled")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+mkdir -p ${libdir}
+cd $WORKSPACE/srcdir/scalapack-2.1.0
```
Comment (jeremiahpslewis) [inline S/SCALAPACK32/build_tarballs.jl:15]:
```suggestion
cd $WORKSPACE/srcdir/scalapack-*
```

**Comment by amontoison** [comment]
@giordano Do you have an idea how to fix the issue with `Darwin` platforms ?

**Comment by giordano** [comment]
https://github.com/JuliaPackaging/Yggdrasil/blob/39c397adca039354b8fa13c2fd8514c0f0fd2577/A/ADIOS2/build_tarballs.jl#L32-L39

File: S/SCALAPACK32/build_tarballs.jl [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+mkdir build && cd build
+cmake .. "${CMAKE_FLAGS[@]}"
+
+make -j${nproc} all
+make install
+"""
+
+# OpenMPI and MPICH are not precompiled for Windows
+# Can't get the code to build for PowerPC with libgfortran3
+platforms = expand_gfortran_versions(filter!(p -> !Sys.iswindows(p) && arch(p) != "powerpc64le", supported_platforms()))
```
Comment (jeremiahpslewis) [inline S/SCALAPACK32/build_tarballs.jl:60]:
```suggestion
platforms = expand_gfortran_versions(supported_platforms(; exclude=p -> Sys.iswindows(p) || arch(p) == "powerpc64le"))
```

**Comment by amontoison** [comment]
With `MPICH_jll` v"3.3.2", MacOS finds MPI_Fortran :thinking:

**Comment by amontoison** [comment]
@giordano I tried your fix and it's not solve the issue.
The only solution that I have is to build without experimental platforms and MPICH_jll v"3.3.2".
`mpifort` is not detected anymore since v"3.4.2" (https://github.com/JuliaPackaging/Yggdrasil/pull/3185) on `apple` platforms.

**Comment by amontoison** [comment]
It finally works for all platforms! :tada:

**Comment by giordano** [comment]
Uhm, you must be sure you don't end up having `SCALAPACK` and `SCALAPACK32` in the same environment, otherwise they'd conflict in an unpredictable way

**Comment by amontoison** [comment]
The final goal is to recompile MUMPS@5 with `SCALAPACK32`.
What did you do to avoid conflicts with `OpenBLAS` and `OpenBLAS32` ?

**Comment by amontoison** [comment]
> Libraries have different file names and sonames.

Cab we just rename the library `libscalapack32` after the compilation ?
```
products = [
    LibraryProduct("libscalapack32", :libscalapack32),
]```

**Comment by giordano** [comment]
Yes, but you also have to set their sonames appropriately, the file name alone does little

**Comment by amontoison** [comment]
Ok, thanks @giordano !
I changed the `name` and the `soname`.

**Pull request merged**